### ISSUE TITLE
Bogus error CS0170 reported for deconstruction in foreach loop

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#17518]
Sync to PR https://github.com/dotnet/roslyn/pull/17519.

Inline variable ```t``` in [code](https://github.com/dotnet/roslyn/pull/17519/commits/6e534740daff0c5462e3c6bc74aa09b72cabf534?diff=split#diff-9236a68bf1a2701b994b2d9d86eece83R242):

```C#
foreach (var t in locals)
{
   var (local, scope) = t;
   ...
   builder.WriteInt32(scope.StartOffset);
   builder.WriteInt32(scope.EndOffset);
   ...
}
```

like so:
```C#
foreach (var (local, scope) in locals)
{
  ...
  builder.WriteInt32(scope.StartOffset);
  builder.WriteInt32(scope.EndOffset);
  ...
}
```

Build Compilers.sln

Actual:
```
src\Compilers\Core\Portable\PEWriter\CustomDebugInfoWriter.cs(259,36,259,53): error CS0170: Use of possibly unassigned field 'StartOffset'
src\Compilers\Core\Portable\PEWriter\CustomDebugInfoWriter.cs(260,36,260,51): error CS0170: Use of possibly unassigned field 'EndOffset'
```

Expected:
No errors.

Unfortunately, I was not able to extract a simpler repro.

**Comment by tmat** [comment]
@agocke FYI - this is the issue I mentioned earlier today.

**Comment by jcouv** [comment]
Sounds like https://github.com/dotnet/roslyn/pull/16318 (fixed in RC3)
FYI @VSadov

**Comment by jcouv** [comment]
Confirmed that the issue no longer repros. I'll remove the workaround in `CustomDebugInfoWriter` (with https://github.com/dotnet/roslyn/pull/18381).

**Comment by jcouv** [comment]
The error did repro in some of the CI legs (Ubuntu). Re-opening.
https://ci.dot.net/job/dotnet_roslyn/job/master/job/ubuntu_14_debug_prtest/1190/

**Comment by jcouv** [comment]
I wasn't able to remove the workaround just yet because the toolset compiler for Ubuntu pre-dates Vlad's fix. We'll have to wait to that toolset to be updated.

**Comment by jcouv** [comment]
The toolset update is coming (https://github.com/dotnet/roslyn/pull/18404)

**Comment by jcouv** [comment]
Latest code in master already removed the workaround. Closing.
